be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -891,13 +947,36 @@ Status BaseBetaRowsetWriter::create_file_writer(uint32_t segment_id, io::FileWri
             fmt::format("failed to create file = {}, file type = {}", segment_path, file_type));
 }
 
+Status BaseBetaRowsetWriter::create_inverted_index_file_writer(
+        uint32_t segment_id, InvertedIndexFileWriterPtr* index_file_writer) {
+    RETURN_IF_ERROR(RowsetWriter::create_inverted_index_file_writer(segment_id, index_file_writer));
+    // used for inverted index format v1
+    (*index_file_writer)->set_file_writer_opts(_context.get_file_writer_options());
+    return Status::OK();
+}
+
 Status BetaRowsetWriter::_create_segment_writer_for_segcompaction(
         std::unique_ptr<segment_v2::SegmentWriter>* writer, int64_t begin, int64_t end) {
     DCHECK(begin >= 0 && end >= 0);
     std::string path = BetaRowset::local_segment_path_segcompacted(_context.tablet_path,
                                                                    _context.rowset_id, begin, end);
     io::FileWriterPtr file_writer;
     RETURN_IF_ERROR(_create_file_writer(path, file_writer));
+    std::string prefix = std::string {InvertedIndexDescriptor::get_index_file_path_prefix(path)};
+    std::string index_path = InvertedIndexDescriptor::get_index_file_path_v2(prefix);
+
+    InvertedIndexFileWriterPtr index_file_writer;
+    if (_context.tablet_schema->has_inverted_index()) {
+        io::FileWriterPtr idx_file_writer;
+        if (_context.tablet_schema->get_inverted_index_storage_format() !=
+            InvertedIndexStorageFormatPB::V1) {
+            RETURN_IF_ERROR(_create_file_writer(index_path, idx_file_writer));
+        }
+        index_file_writer = std::make_unique<InvertedIndexFileWriter>(
+                _context.fs(), path, _context.rowset_id.to_string(), _num_segcompacted,
+                _context.tablet_schema->get_inverted_index_storage_format(),
+                std::move(idx_file_writer));
```

Comment:
double move?

Reply:
`std::move()` only marks this value as an rvalue.